backpack/extensions/firstorder/batch_l2_grad/convnd.py

```diff
@@ -1,23 +1,62 @@
+"""batch_l2 extension for Conv."""
 from torch import einsum
 
-from backpack.extensions.firstorder.base import FirstOrderModuleExtension
+from backpack.core.derivatives.convnd import ConvNDDerivatives
+from backpack.extensions.firstorder.batch_l2_grad.batch_l2_base import BatchL2Base
 from backpack.utils import conv as convUtils
 
 
-class BatchL2ConvND(FirstOrderModuleExtension):
+class BatchL2ConvND(BatchL2Base):
+    """batch_l2 extension for Conv."""
+
     def __init__(self, N, params=None):
-        super().__init__(params=params)
+        """Initialization.
+
+        Args:
+            N: number of dimensions
+            params: list of parameter names. Defaults to None.
+        """
         self.N = N
+        super().__init__(params=params, derivatives=ConvNDDerivatives(N=self.N))
```

Comment:
`N` seems to only be used to determine which derivative should be used -> Replace (`ConvNDDerivatives` + `N`) by `Conv{1,2,3}d`?

This would also make the constructor unnecessary.

Reply:
**Update:** The above comment is incorrect. as `self.N` is used further below as argument of `convUtils.get_weight_gradient_factors`. Could you take a look how difficult it would be to eliminate `N` from the arguments of `convUtils.get_weight_gradient_factors`? I recently did a very similar refactoring on those files, however for the second-order, rather than the first-order extensions.

Reply:
Refactored similar to convtransposednd.py.